### PR TITLE
IGVF-862-dbxref-reference-files

### PR DESCRIPTION
**Pull request by ian-whaling**

### CODE REVIEW

File: src/igvfd/upgrade/file.py [hunk 1]
```diff
@@ -57,3 +57,13 @@ def sequence_file_3_4(value, system):
...
         value['notes'] = notes
+
+
+@upgrade_step('reference_file', '4', '5')
+@upgrade_step('matrix_file', '1', '2')
+@upgrade_step('signal_file', '1', '2')
+@upgrade_step('configuration_file', '1', '2')
+@upgrade_step('alignment_file', '1', '2')
+def file_4_5(value, system):
+    if 'dbxrefs' in value:
```
Comment (gabdank) [inline src/igvfd/upgrade/file.py:68]: should not this step be true ONLY if dbxref is an empty string?

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/upgrade/file.py (hunk 1 above)
Comment (ian-whaling) [reply src/igvfd/upgrade/file.py:68]: these files atp can only have dbxrefs if it consists of empty string(s)

**Review by gabdank — DISMISSED** [review]

File: src/igvfd/upgrade/file.py [hunk 2]
```diff
@@ -57,3 +57,18 @@ def sequence_file_3_4(value, system):
         else:
             notes = f'The mean read length previously exceeded the upper limit of 300000000 so was set to 300000000, please assign the appropriate length.'
         value['notes'] = notes
+
+
+@upgrade_step('reference_file', '4', '5')
+@upgrade_step('matrix_file', '1', '2')
+@upgrade_step('signal_file', '1', '2')
+@upgrade_step('configuration_file', '1', '2')
+@upgrade_step('alignment_file', '1', '2')
```
Comment (gabdank) [inline src/igvfd/upgrade/file.py:66]: i think the schema versions here are incorrect, for example sequence_file is 4->5

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/upgrade/file.py (hunk 2 above)
Comment (ian-whaling) [reply src/igvfd/upgrade/file.py:66]: fixed

File: src/igvfd/schemas/changelogs/file.md [hunk 3]
```diff
@@ -1,5 +1,4 @@
 ## Changelog for *`file.json`*
 
-### Release v1
-
-* Deprecate `reference_data` and `sequence_data` schemas.
+* Disallow submission of empty string in `dbxrefs`. (11/01/2023)
```
Comment (iyoungworth) [inline src/igvfd/schemas/changelogs/file.md:3]: Should update this to next version date

**Review by iyoungworth — CHANGES_REQUESTED** [review]
General request for concrete file schemas: if minItems restriction is new to all cases, I think each changelog should mention that too

For file schemas without their own dbxrefs pattern (all but Reference and Sequence): please add a dbxrefs section with comment indicating it isn't currently submittable (a specific pattern isn't present, so nothing will match File regex pattern "(?!)"

File: src/igvfd/schemas/changelogs/file.md (hunk 3 above)
Comment (ian-whaling) [reply src/igvfd/schemas/changelogs/file.md]: updated

File: src/igvfd/schemas/genome_browser_annotation_file.json [hunk 4]
```diff
@@ -170,6 +170,9 @@
                 "appscript": "GRCh38",
                 "igvf_utils": "GRCh38"
             }
+        },
```
Comment (iyoungworth) [inline src/igvfd/schemas/genome_browser_annotation_file.json:173]: The schema version needs to be bumped

File: src/igvfd/schemas/tabular_file.json [hunk 5]
```diff
@@ -178,6 +178,9 @@
                 "appscript": "GRCh38",
                 "igvf_utils": "GRCh38"
             }
+        },
```
Comment (iyoungworth) [inline src/igvfd/schemas/tabular_file.json:181]: The schema version needs to be bumped up

**Review by iyoungworth — CHANGES_REQUESTED** [review]
New dbxrefs comments and changelogs look great, but the 2 new concrete File classes need default schema version bumped to '2'

**Review by iyoungworth — APPROVED** [review]

**Pull request merged**